components/RichText.tsx

```diff
@@ -0,0 +1,95 @@
+import { documentToReactComponents } from '@contentful/rich-text-react-renderer';
+import { BLOCKS, MARKS, INLINES } from '@contentful/rich-text-types';
+import React from 'react';
+import styles from 'styles/RichText.module.scss';
+
+type Theme = 'LIGHT' | 'DARK' | 'INDIGO';
+
+type RichTextProps = {
+    theme: Theme;
+    content: any;
+};
+
+const renderParagraph = (theme: Theme, children: React.ReactNode) => {
+    switch (theme) {
+        case 'LIGHT':
+            return <p className={styles.paragraphLight}>{children}</p>;
+        case 'DARK':
+            return <p className={styles.paragraphDark}>{children}</p>;
+        case 'INDIGO':
+            return <p className={styles.paragraphIndigo}>{children}</p>;
+    }
+};
+
+const renderLink = (theme: Theme, node: any, children: React.ReactNode) => {
+    switch (theme) {
+        case 'LIGHT':
+            return (
+                <a className={styles.hyperlink} href={node.data.uri}>
+                    {children}
+                </a>
+            );
+        case 'DARK':
+            return (
+                <a className={styles.hyperlinkDark} href={node.data.uri}>
+                    {children}
+                </a>
+            );
+        case 'INDIGO':
+            return (
+                <a className={styles.hyperlink} href={node.data.uri}>
+                    {children}
+                </a>
+            );
+    }
+};
+
+const getRenderOptions = (theme: Theme) => {
+    return {
+        renderMark: {
+            [MARKS.BOLD]: (text: React.ReactNode) => (
+                <span className={styles.bold}>{text}</span>
+            ),
+            [MARKS.ITALIC]: (text: React.ReactNode) => (
+                <span className={styles.italic}>{text}</span>
+            ),
+            [MARKS.UNDERLINE]: (text: React.ReactNode) => (
+                <span className={styles.underline}>{text}</span>
+            ),
+        },
+        renderNode: {
```

Comment:
Sorry James, I'll have to review this another time to make sure I understand it properly before giving feedback, will look into this again tomorrow when i have more time.